### PR TITLE
feat (sdk): parse workflow as code

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: sdk/exportentities/workflow.go [hunk 1]
```diff
@@ -153,3 +159,271 @@ func NewWorkflow(w sdk.Workflow, withPermission bool) (Workflow, error) {
 
 	return exportedWorkflow, nil
 }
+
+// Entries returns the map of all workflow entries
+func (w Workflow) Entries() map[string]WorkflowEntry {
```
Comment (sguiheux) [inline sdk/exportentities/workflow.go]: WOrkflowEntry or NodeEntry?

Comment (fsamin) [reply sdk/exportentities/workflow.go]: WorkflowEntry => https://github.com/ovh/cds/blob/master/sdk/exportentities/workflow.go#L21

File: sdk/workflow.go [hunk 2]
```diff
@@ -126,6 +126,32 @@ func (w *Workflow) Nodes(withRoot bool) []WorkflowNode {
 	return res
 }
 
+func (w *Workflow) AddTrigger(name string, dest WorkflowNode) {
```
Comment (yesnault) [inline sdk/workflow.go]: doc

File: sdk/workflow.go [hunk 3]
```diff
@@ -126,6 +126,32 @@ func (w *Workflow) Nodes(withRoot bool) []WorkflowNode {
...
+
+	w.Root.AddTrigger(name, dest)
+	for i := range w.Joins {
+		for j := range w.Joins[i].Triggers {
+			w.Joins[i].Triggers[j].WorkflowDestNode.AddTrigger(name, dest)
+		}
+	}
+}
+
+func (n *WorkflowNode) AddTrigger(name string, dest WorkflowNode) {
```
Comment (yesnault) [inline sdk/workflow.go]: doc

**Pull request merged**